cfme/tests/cloud/test_keypairs.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from cfme.fixtures import pytest_selenium as sel
+from cfme.cloud.keypairs import KeyPair
+from utils import testgen
+from cfme.web_ui import ButtonGroup, form_buttons
+
+pytestmark = [
+    pytest.mark.ignore_stream("upstream"),
+]
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.provider_by_type(
+        metafunc, ['ec2'])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope='module')
+
+
+def set_grid_view(name):
+    bg = ButtonGroup(name)
+    sel.force_navigate("my_settings_default_views")
+    default_view = bg.active
+    if default_view != 'Grid View':
+        bg.choose('Grid View')
+        sel.click(form_buttons.save)
+
+
+@pytest.fixture(scope="function")
+def keypairs(setup_provider, provider, provisioning):
+    set_grid_view("KeyPairs")
+    keypairs_name = provisioning['keypairs']
+    keypairs = KeyPair(keypairs_name)
+    return keypairs
+
+
+@pytest.mark.tier(3)
+def test_parameters_link(keypairs):
+    keypairs.nav_to_parameters_link()
+
+
+@pytest.mark.tier(3)
+def test_delete(keypairs):
```

Comment:
It should be named test_keypair_delete or something like that. I would rather see some test_keypair_crud if that is possible, because you need to create a keypair for the test to tinker with.

Btw you need to add some comments for the tests.